### PR TITLE
Light: add testcase for throttle() filter

### PR DESCRIPTION
**Pull request by alltilla**
Signed-off-by: Attila Szakacs <attila.szakacs@oneidentity.com>

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: tests/python_functional/functional_tests/filters/throttle/test_throttle_filter.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    for s_network in network_sources:
+        s_network.write_log(input_messages, rate=eps)
+
+
+def wait_for_constant_msg_flow():
+    time.sleep(1)
+
+
+def is_outgoing_eps_throttled(throttle_filters, throttle_rate, key_resolutions, tolerance):
+    measurement_duration = 1  # seconds
```
[the remark this answers is not in the document]
Comment (alltilla) [reply tests/python_functional/functional_tests/filters/throttle/test_throttle_filter.py]: This testcase was replaced by another one, written by Micek.

File: tests/python_functional/functional_tests/filters/throttle/test_throttle_filter.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        start_msg_counts.append(msg_count)
+
+    time.sleep(measurement_duration)
+
+    for i, f_throttle in enumerate(throttle_filters):
+        msg_count = f_throttle.get_stats()["matched"]
+        outgoing_rate = (msg_count - start_msg_counts[i]) / measurement_duration
+
+        if not math.isclose(outgoing_rate, expected_rate, rel_tol=tolerance):
+            raise Exception(
```
[the remark this answers is not in the document]
Comment (alltilla) [reply tests/python_functional/functional_tests/filters/throttle/test_throttle_filter.py]: This testcase was replaced by another one, written by Micek.

File: tests/python_functional/functional_tests/filters/throttle/test_throttle_filter.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+def is_outgoing_eps_throttled(throttle_filters, throttle_rate, key_resolutions, tolerance):
+    measurement_duration = 1  # seconds
+    expected_rate = throttle_rate * key_resolutions
+    start_msg_counts = []
+
+    for f_throttle in throttle_filters:
+        msg_count = f_throttle.get_stats()["matched"]
+        start_msg_counts.append(msg_count)
+
+    time.sleep(measurement_duration)
```
[the remark this answers is not in the document]
Comment (alltilla) [reply tests/python_functional/functional_tests/filters/throttle/test_throttle_filter.py]: This testcase was replaced by another one, written by Micek.

File: tests/python_functional/functional_tests/filters/throttle/test_throttle_filter.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+def are_messages_received_from_all_key_resolutions(file_destinations, key_resolutions, tolerance):
+    for d_file in file_destinations:
+        msg_counts = {}
+
+        for program_idx in range(key_resolutions):
+            program_field = "test_program_{}".format(program_idx)
+            msg_count = count_message_in_file(d_file.get_path(), program_field)
+            msg_counts[program_field] = msg_count
+
+        least_messages = min(msg_counts.items(), key=lambda k: k[1])
```
[the remark this answers is not in the document]
Comment (alltilla) [reply tests/python_functional/functional_tests/filters/throttle/test_throttle_filter.py]: This testcase was replaced by another one, written by Micek.

File: tests/python_functional/functional_tests/filters/throttle/test_throttle_filter.py [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+                "least_messages: {}, most_messages: {}".format(least_messages, most_messages),
+            )
+
+    return True
+
+
+def test_throttle_filter(config, syslog_ng, port_allocator, bsd_formatter):
+    config.update_global_options(stats_level=1)
+
+    s_network_1, f_throttle_1, d_file_1 = create_throttled_logpath(config, port_allocator, THROTTLE_RATE)
```
[the remark this answers is not in the document]
Comment (alltilla) [reply tests/python_functional/functional_tests/filters/throttle/test_throttle_filter.py]: Yes this was unnecessary. This testcase was replaced by another one, written by Micek.

**Review by mitzkia — APPROVED** [review]
Thanks for the changes. Approved

**Comment by mitzkia** [comment]
One small thing, could you please add new files (for example testcase) to the Makefile.am

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Review by mitzkia — APPROVED** [review]
Thanks for the change. Approve

**Review by Kokan — APPROVED** [review]

**Pull request merged**